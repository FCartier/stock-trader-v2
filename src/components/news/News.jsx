import React from "react";

import { SectionHeader } from "../reusables/SectionHeader";


const News = props => {
  const { news } = props;

  return news.length ? (
    <div>
      <div className="news">
        <SectionHeader title="NEWS" />
        <ul className="news-list">
          {news.map((article, index) => {
            return (
              <li key={index}>
                <p className="article-header">
                  <a href={article.url}>{article.headline}</a>
                </p>
                <p className="article-footer">{article.source}</p>
              </li>
            );
          })}
        </ul>
      </div>
    </div>
  ) : null;
};

export default News;
