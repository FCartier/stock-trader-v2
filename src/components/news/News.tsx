import * as React from "react";

import { SectionHeader } from "../reusables/SectionHeader.tsx";
import { NewsItem, NewsList } from './types';

const News: React.SFC<NewsList> = ({news}) => {
  console.log(news);


  return renderCondition(news) ? (
    <div>
      <div className="news">
        <SectionHeader title="NEWS" />
        <ul className="news-list">
          {news
          .slice(0, 5)
          .map((article: NewsItem, index: number) => {
            return (
              <li key={index}>
                <p className="article-header">
                  <a href={article.url}>{article.headline}</a>
                </p>
                <p className="article-footer">{article.source}</p>
              </li>
            );
          })}
        </ul>
      </div>
    </div>
  ) : null;
};

function renderCondition(news: []) {
  console.log(news.length)
  if (news.length > 0) {
    return true
  }
}

export default News;
