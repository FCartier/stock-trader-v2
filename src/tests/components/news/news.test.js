import React from 'react';
import News from '../../../components/news/News';
import renderer from 'react-test-renderer';
import testData from './news.data.json'


describe("news component", () => {
  test('news component should display all data fields', () => {
    const component = renderer.create(
      <News news={testData}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})
