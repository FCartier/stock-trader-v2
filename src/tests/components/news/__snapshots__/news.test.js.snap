// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`news component news component should display all data fields 1`] = `
<div>
  <div
    className="news"
  >
    <div
      className="category-header"
    >
      <h3>
        NEWS
      </h3>
    </div>
    <ul
      className="news-list"
    >
      <li>
        <p
          className="article-header"
        >
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/6930494542254354"
          >
            China's alleged tiny-chip spying on US tech giants was 'pretty amateurish,' says ex-Senate advisor
          </a>
        </p>
        <p
          className="article-footer"
        >
          CNBC
        </p>
      </li>
      <li>
        <p
          className="article-header"
        >
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/8254196132130335"
          >
            Nicholas Ward's Dividend Growth Portfolio: Third Quarter Update
          </a>
        </p>
        <p
          className="article-footer"
        >
          SeekingAlpha
        </p>
      </li>
      <li>
        <p
          className="article-header"
        >
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/7955116944851206"
          >
            Stocks making the biggest moves premarket: SNAP, TSLA, AAPL, FB & more
          </a>
        </p>
        <p
          className="article-footer"
        >
          CNBC
        </p>
      </li>
      <li>
        <p
          className="article-header"
        >
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/4910314825150120"
          >
            Markets Anticipate September Jobs Report (Wall Street Breakfast Podcast)
          </a>
        </p>
        <p
          className="article-footer"
        >
          SeekingAlpha
        </p>
      </li>
      <li>
        <p
          className="article-header"
        >
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/5126119929706806"
          >
            The Selloff Has Begun - Cramer's Mad Money (10/4/18)
          </a>
        </p>
        <p
          className="article-footer"
        >
          SeekingAlpha
        </p>
      </li>
      <li>
        <p
          className="article-header"
        >
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/8269496580256224"
          >
            Stock futures seesaw ahead of Friday's jobs report
          </a>
        </p>
        <p
          className="article-footer"
        >
          CNBC
        </p>
      </li>
      <li>
        <p
          className="article-header"
        >
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/6477702170575935"
          >
            Wall Street Breakfast: Ready For September Jobs Report
          </a>
        </p>
        <p
          className="article-footer"
        >
          SeekingAlpha
        </p>
      </li>
      <li>
        <p
          className="article-header"
        >
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/7267303021565616"
          >
            Q3 Portfolio Review
          </a>
        </p>
        <p
          className="article-footer"
        >
          SeekingAlpha
        </p>
      </li>
      <li>
        <p
          className="article-header"
        >
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/5712477302901783"
          >
            September Dividend Income Report
          </a>
        </p>
        <p
          className="article-footer"
        >
          SeekingAlpha
        </p>
      </li>
      <li>
        <p
          className="article-header"
        >
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/6695808814944607"
          >
            Chinese spy chip report weighs on Apple suppliers
          </a>
        </p>
        <p
          className="article-footer"
        >
          SeekingAlpha
        </p>
      </li>
      <li>
        <p
          className="article-header"
        >
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/8258769712992531"
          >
            Apple suppliers' shares fall on the back of report alleging hidden Chinese spy chips
          </a>
        </p>
        <p
          className="article-footer"
        >
          CNBC
        </p>
      </li>
      <li>
        <p
          className="article-header"
        >
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/8295244546945973"
          >
            Lenovo shares tumble about 15 percent following report about alleged Chinese spy chips
          </a>
        </p>
        <p
          className="article-footer"
        >
          CNBC
        </p>
      </li>
      <li>
        <p
          className="article-header"
        >
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/8675697439173982"
          >
            Does Facebook Fit The Buffett Criteria?
          </a>
        </p>
        <p
          className="article-footer"
        >
          SeekingAlpha
        </p>
      </li>
    </ul>
  </div>
</div>
`;
